testing/unittests/dalitests.cpp

```diff
@@ -2276,10 +2276,16 @@ class CDaliDFSStressTests : public CppUnit::TestFixture
             e->Release();
         }
 
-        // To make sure renamed files are cleaned properly
-        printf("Renaming 'regress::rename::other2 to 'sub2' on auto-commit\n");
-        dir.renamePhysical("regress::rename::other2", "regress::rename::sub2", user, transaction);
-        ASSERT(dir.exists("regress::rename::sub2", user, true, false) && "Renamed from other2 failed");
+        try {
+            // To make sure renamed files are cleaned properly
+            printf("Renaming 'regress::rename::other2 to 'sub2' on auto-commit\n");
+            dir.renamePhysical("regress::rename::other2", "regress::rename::sub2", user, transaction);
+            ASSERT(dir.exists("regress::rename::sub2", user, true, false) && "Renamed from other2 failed");
+        }
+        catch  (IException *e) {
+            // Expecting exception
```

Comment:
Why is it expected?

```
rename: could not rename logical file regress::rename::other2 to regress::rename::sub2: [  2: CFile::move(/var/lib/HPCCSystems/hpcc-data/regress/rename/other2._1_of_1, /var/lib/HPCCSystems/hpcc-data/regress/rename/sub2._1_of_1), No such file or directory] 
```

I can' t see why this exception sould happen. As far as I can see other2 should exist, so the rename should have succeeded.

And, for unexpected exceptions (to avoid 'uncaught exception of unknown type'), it would be better to capture any that may occur in the whole function, capture the text of the exception and output it with a CPPUNIT_FAIL - so the reason for unexpected failure was clear.